src/core/ledger/ledger.js

```diff
@@ -768,6 +774,79 @@ export class Ledger {
     account.payoutAddresses.delete(JSON.stringify(currencyResult.value));
   }
 
+  /**
+   * no-ops if integration tracking is already enabled
+   */
+  enableIntegrationTracking(): Ledger {
+    if (!this._trackIntegration) {
+      this._createAndProcessEvent({
+        type: "TOGGLE_INTEGRATION",
+      });
+    }
+    return this;
+  }
+
+  /**
+   * no-ops if integration tracking is already disabled
+   */
+  disableIntegrationTracking(): Ledger {
+    if (this._trackIntegration) {
+      this._createAndProcessEvent({
+        type: "TOGGLE_INTEGRATION",
+      });
+    }
+    return this;
+  }
+  _toggleIntegrationTracking(_: ToggleIntegration) {
+    this._trackIntegration = !this._trackIntegration;
+    if (!this._trackIntegration) this._integrationDistributions.clear();
+  }
+
+  /**
+   * Mark integrations as executed.
+   * Validates that:
+   * a) executions are tracked
+   * b) executions are not yet marked as executed
+   * c) distribution tracking is enabled
+   */
+  runIntegration(id: DistributionId): Ledger {
+    if (!this._trackIntegration) {
+      throw new Error("integration tracking not enabled");
+    }
+    if (this._integrationDistributions.get(id) === undefined) {
+      throw new Error("Distribution not eligible to executed");
+    }
+    if (this._integrationDistributions.get(id) === true) {
+      throw new Error("Integration has already executed this distribution");
+    }
+    this._createAndProcessEvent({
+      type: "RUN_INTEGRATION",
+      id,
+    });
+    return this;
+  }
+  _runIntegration({id}: RunIntegration) {
+    this._integrationDistributions.set(id, true);
+  }
+
+  /**
+   * Is cleared each time `disableIntegrationTracking` is called
+   */
+  trackedDistributions(): Iterator<DistributionId> {
+    return this._integrationDistributions.keys();
+  }
+
+  /**
+   * Returns the status of a tracked distribution
+   * If the distribution has not been executed by the integration, `false` is
+   * returned.
+   * If the distribution has been executed, `true` is returned.
+   * If the distribution is untracked, `undefined` is returned.
+   */
+  integrationExecuted(id: DistributionId): ?boolean {
```

Comment:
`isIntegrationExecuted` - boolean prefixes like "is" or "has" go a long way for code readability.